launchdarkly-server-sdk/src/migrations/migrator.rs

```diff
@@ -0,0 +1,317 @@
+// TODO: Remove this when subsequent PRs have added the required implementations.
+#![allow(dead_code)]
+
+use launchdarkly_server_sdk_evaluation::Context;
+
+use crate::{Client, ExecutionOrder, Origin, Stage};
+
+/// An internally used struct to represent the result of a migration operation along with the
+/// origin it was executed against.
+pub struct MigrationResultPair {
+    value: serde_json::Value,
+    origin: Origin,
+}
+
+/// MigrationResult represents the result of a migration operation. If the operation was
+/// successful, the result will contain a pair of values representing the result of the operation
+/// and the origin it was executed against. If the operation failed, the result will contain an
+/// error.
+type MigrationResult = Result<MigrationResultPair, Box<dyn std::error::Error>>;
+
+/// A write result contains the operation results against both the authoritative and
+/// non-authoritative origins.
+///
+/// Authoritative writes are always executed first. In the event of a failure, the
+/// non-authoritative write will not be executed, resulting in a `None` value in the final
+/// MigrationWriteResult.
+pub struct MigrationWriteResult {
+    authoritative: MigrationResult,
+    nonauthoritative: Option<MigrationResult>,
+}
+
+// MigrationComparisonFn is used to compare the results of two migration operations. If the
+// provided results are equal, this method will return true and false otherwise.
+type MigrationComparisonFn = fn(serde_json::Value, serde_json::Value) -> bool;
+
+// MigrationImplFn represents the customer defined migration operation function. This method is
```

Comment:
```suggestion
// MigrationImplFn represents the user defined migration operation function. This method is
```

(I see `user` used in other comments)